Public/Src/Cache/ContentStore/Interfaces/Results/PutResult.cs

```diff
@@ -107,5 +107,13 @@ public override string ToString()
                 ? $"Success Hash={ContentHash.ToShortString()} Size={ContentSize}{this.GetDiagnosticsMessageForTracing()}"
                 : GetErrorString();
         }
+
+        internal class OptionalMetadata

```

Comment:
I don't think that the type name should reflect that the data is optional.

When the property is mutable it kind of implies that the data is optional.

I think you can name it ExtraMetadata.